src/invaders.rs

```diff
@@ -21,48 +23,54 @@ pub struct Invaders {
 impl Invaders {
     pub fn new() -> Self {
         let mut army = Vec::new();
-        for x in 0..NUM_COLS {
-            for y in 0..NUM_ROWS {
-                if (x > 1)
-                    && (x < NUM_COLS - 2)
-                    && (y > 0)
-                    && (y < 9)
-                    && (x % 2 == 0)
-                    && (y % 2 == 0)
-                {
-                    army.push(Invader { x, y, points: 1 });
-                }
+        let mut faces_list = INVADER_FACES.iter();
+        for y in (INVADERS_FIRST_ROW..=INVADERS_LAST_ROW).step_by(INVADERS_ROW_INTERVAL) {
```

Comment:
Oooo, your idea to use `step_by` is quite nice!

I see that you're trying to iterate through the `y` values and the faces at the same rate, and resorted to manual iteration for the faces. Let's iterate through both at once with [zip](https://doc.rust-lang.org/std/iter/fn.zip.html)!

```suggestion
        for (y, faces) in (INVADERS_FIRST_ROW..=INVADERS_LAST_ROW)
            .step_by(INVADERS_ROW_INTERVAL)
            .zip(INVADER_FACES.iter())
        {
```

Reply:
I did read about Zip, but never gave it a thought here. But it is a very elegant and self-describing method